support/cas-server-support-oauth-core-api/src/main/java/org/apereo/cas/support/oauth/authenticator/OAuth20ProofKeyCodeExchangeAuthenticator.java

```diff
@@ -29,20 +30,34 @@
  */
 @Slf4j
 public class OAuth20ProofKeyCodeExchangeAuthenticator extends OAuth20ClientIdClientSecretAuthenticator {
-    private final TicketRegistry ticketRegistry;
 
     public OAuth20ProofKeyCodeExchangeAuthenticator(final ServicesManager servicesManager,
                                                     final ServiceFactory webApplicationServiceFactory,
                                                     final AuditableExecution registeredServiceAccessStrategyEnforcer,
                                                     final TicketRegistry ticketRegistry,
                                                     final CipherExecutor<Serializable, String> registeredServiceCipherExecutor) {
-        super(servicesManager, webApplicationServiceFactory, registeredServiceAccessStrategyEnforcer, registeredServiceCipherExecutor);
-        this.ticketRegistry = ticketRegistry;
+        super(servicesManager, webApplicationServiceFactory, registeredServiceAccessStrategyEnforcer, registeredServiceCipherExecutor, ticketRegistry);
+    }
+
+    /**
+     * Can perform authentication if the code_verifier is supplied.
+     * @param context the context
+     * @return
+     */
+    @Override
+    protected boolean canAuthenticate(final WebContext context) {
```

Comment:
While this is fine, in the future we could likely refactor this to be a generic `supports(Context)` method instead.